src/marshmallow_sqlalchemy/convert.py

```diff
@@ -348,7 +348,7 @@ def _add_relationship_kwargs(self, kwargs, prop):
         nullable = True
         for pair in prop.local_remote_pairs:
             if not pair[0].nullable:
-                if prop.uselist is True:
+                if not self.DIRECTION_MAPPING[prop.direction.name]:
```

Comment:
This should be handling 2 cases:
1. For anything which has a list - we expect an empty list - and so nullable=False
2. And for anything which has a FK on the table itself - we should consider the nullable from the FK (MANYTOONE case)

```
if not pair[0].nullable:
    if prop.uselist is True or self.DIRECTION_MAPPING[prop.direction.name] is False:
        nullable = False
continue
```

MANYTOONE case:
```py
class Book:
    ...
    author_id = Column(Integer, ForeignKey('author.id'), nullable=False)
    author = relationship('Author', lazy='selectin')
```

MANYTOMANY case with uselist=True:
```py
class Book:
    ...
    author = relationship('Author', secondary=book_author, lazy='selectin', uselist=True)
```

MANYTOMANY case with uselist=False:
```py
class Book:
    ...
    author = relationship('Author', secondary=book_author, lazy='selectin', uselist=False)
```


Can we add testcases for various direction X uselist scenarios ?

Reply:
Hi @AbdealiJK , 

Have updated the changes based on the feedback in comment